src/correction.cc

```diff
@@ -197,10 +213,11 @@ Formula::Formula(const JSONObject& json, const Correction& context, bool generic
 
   std::vector<size_t> variableIdx;
   for (const auto& item : json.getRequired<rapidjson::Value::ConstArray>("variables")) {
-    auto idx = context.input_index(item.GetString());
-    if ( context.inputs()[idx].type() != Variable::VarType::real ) {
+    auto idx = std::distance(std::begin(inputs),
```

Comment:
This replaces the implementation of `Correction::input_index`. I'm fine with being more specific about what is needed to create a formula, but note that in that function an exception is raised if no input with the given name exists. Here I guess there is no error handling for invalid names?
Since `Correction::input_index` is used in several places and in most of them we just want to find the index in the `std::vector<Variable>`, perhaps this should be a free function that takes the input list and does the find?

Reply:
thank you, that's an oversight!

Reply:
added a new commit to address this